src/target/riscv_debug.c

```diff
@@ -304,6 +304,30 @@ void riscv_dmi_init(riscv_dmi_s *const dmi)
 	do {
 		/* Read out the DM's status register */
 		uint32_t dm_status = 0;
+		uint32_t dm_control = 0;
+		bool dm_active = false;
+		static uint8_t counter = 0;
+
+		/* Turn on DM before trying to read version  */
+		if (!riscv_dmi_write(dmi, base_addr + RV_DM_CONTROL, RV_DM_CTRL_ACTIVE)) {
+			DEBUG_ERROR("error turning on DM!\n");
+			return;
+		}
+
+		/* After changing the value of dm_active, the debugger must poll dmcontrol
+		 * until dm_active has taken the requested value */
+		while (dm_active == false) {
```

Comment:
`bool` is a true boolean type - this check can and should be simplified to `while (!dm_active)`